calculator/src/main/java/com/market/banica/calculator/dto/ProductPriceComponentsSet.java

```diff
@@ -0,0 +1,48 @@
+package com.market.banica.calculator.dto;
+
+import com.market.banica.calculator.model.Pair;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Data
+@NoArgsConstructor
+public class ProductPriceComponentsSet implements Comparable<ProductPriceComponentsSet> {
+
+    private static final AtomicInteger count = new AtomicInteger(0);
+
+    @EqualsAndHashCode.Exclude
+    private final int productId = count.incrementAndGet();
+
+    private String productName;
+
+    private BigDecimal price;
+
+    private Pair<Long,Long> reservedQuantityRange_Start_End = new Pair<>(0L,0L);
+
```

Comment:
Wouldn't be better without underscores in the name?

Reply:
The point was to emphasise what the Pair components are. It was applied for code readability, I agree it is not consistent with the team code style. Fixed